app/models/preference.rb

```diff
@@ -4,4 +4,36 @@
 class Preference < ActiveRecord::Base
   belongs_to :user
   belongs_to :section
+
+  validates_presence_of :user
+  validates_presence_of :section
```

Comment:
Prefer the new style validations `validates :column, presence: value` over `validates_presence_of`.